frontend/src/features/public-form/utils/createSubmission.ts

```diff
@@ -1,13 +1,52 @@
-import { forOwn, isEmpty } from 'lodash'
+import { encode as encodeBase64 } from '@stablelib/base64'
+import { chain, forOwn, isEmpty, omit } from 'lodash'
 
 import { BasicField, FormFieldDto } from '~shared/types/field'
-import { FieldResponse } from '~shared/types/response'
+import { EmailResponse, FieldResponse } from '~shared/types/response'
+import {
+  StorageModeAttachment,
+  StorageModeAttachmentsMap,
+  StorageModeSubmissionContentDto,
+} from '~shared/types/submission'
 
+import formsgSdk from '~utils/formSdk'
 import { AttachmentFieldSchema } from '~templates/Field'
 
 import { transformInputsToOutputs } from './inputTransformation'
 import { validateAttachmentInput } from './inputValidation'
 
+// The current encrypt version to assign to the encrypted submission.
+// This is needed if we ever break backwards compatibility with
+// end-to-end encryption
+const ENCRYPT_VERSION = 1
+
+export const createEncryptedSubmissionData = async (
+  formFields: FormFieldDto[],
+  formInputs: Record<string, unknown>,
+  publicKey: string,
+): Promise<StorageModeSubmissionContentDto> => {
+  const responses = createResponsesArray(formFields, formInputs)
+  const encryptedContent = formsgSdk.crypto.encrypt(responses, publicKey)
```

Comment:
didn't we specifically create an [`EncryptionService`](https://github.com/opengovsg/FormSG/blob/develop/src/public/services/EncryptionService.ts) to ensure that there are no bugs with encryption in React? we identified this as the biggest risk of React migration right

Reply:
Brainfart; I will use that instead.

Reply:
Extract `validateResponses` utility function from `EncryptionService`; it can be used to validate both email and storage mode submissions.

In 0f46536